tap_zendesk/streams.py

```diff
@@ -352,11 +384,22 @@ def emit_sub_stream_metrics(sub_stream):
 
         for rec in self._empty_buffer():
             yield rec
+
         emit_sub_stream_metrics(audits_stream)
         emit_sub_stream_metrics(metrics_stream)
         emit_sub_stream_metrics(comments_stream)
         singer.write_state(state)
 
+    def check_access(self):
+        '''
+        Check whether the permission was given to access stream resources or not.
+        '''
+        url = self.endpoint.format(self.config['subdomain'])
+        HEADERS['Authorization'] = 'Bearer {}'.format(self.config["access_token"])
+
+        http.call_api(url, self.config.get('request_timeout', DEFAULT_TIMEOUT), params={'start_time': 1610368140, 'per_page': 1}, headers=HEADERS)
```

Comment:
@prijendev Why is there a static time in the code?

Reply:
Here we calls api to just check api permission and we need to pass required `start_time` parameter. It can be anything. Replaced it with constant